File: src/Tiantong.Iot.Plc/Entities/Plc.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Tiantong.Iot.Plc
{
  [Table("plcs")]
  public class Plc
  {
    public virtual int Id { get; set; } = 0;

    public virtual string Model { get; set; } = "test";

    public virtual string Name { get; set; }

    public virtual string Host { get; set; } = "";

    public virtual int Port { get; set; } = 0;

    public virtual bool IsRunning { get; set; } = false;

    public virtual DateTime CreatedAt { get; set; } = DateTime.Now;

    public List<PlcState> States { get; set; }

  }
}

File: projects/Midos.Iot/Midos.Iot.Client/src/entities/Plc.ts
import DateTime from "@midos/vue-ui/src/utils/DateTime";

export enum PlcModel {
  mc3eBinary = "mc3e-binary",
  mc1eBinary = "mc1e-binary",
  s7200Smart = "s7200smart",
  test = "test",
}

export class Plc {
  id = 0

  name = ""

  number = ""

  model = PlcModel.mc3eBinary

  host = "192.168.1.1"

  port = 8000

  comment = ""

  created_at = DateTime.now
}
